src/controller/AutoCommissioner.cpp

```diff
@@ -310,16 +310,26 @@ CHIP_ERROR AutoCommissioner::CommissioningStepFinished(CHIP_ERROR err, Commissio
     if (err != CHIP_NO_ERROR)
     {
         ChipLogError(Controller, "Failed to perform commissioning step %d", static_cast<int>(report.stageCompleted));
+        if (report.stageCompleted == CommissioningStage::kAttestationVerification &&
+            (report.Get<AdditionalErrorInfo>().attestationResult ==
+                 Credentials::AttestationVerificationResult::kDacProductIdMismatch ||
+             report.Get<AdditionalErrorInfo>().attestationResult ==
+                 Credentials::AttestationVerificationResult::kDacVendorIdMismatch))
```

Comment:
Please parenthesize sub-expressions

Reply:
Can you clarify? There are parentheses around the check for the attestation result || - do you want them also around the check for stage complete? or around each of the attestation result checks?

Reply:
Added a smattering of parentheses.